gravitee-apim-console-webui/src/management/api/documentation-v4/api-documentation-v4.component.ts

```diff
@@ -14,36 +14,49 @@
  * limitations under the License.
  */
 
-import { Component, OnDestroy, OnInit } from '@angular/core';
-import { Subject } from 'rxjs';
+import { Component, computed, OnDestroy, OnInit } from '@angular/core';
+import { Observable, Subject, of, combineLatest, EMPTY, BehaviorSubject } from 'rxjs';
 import { MatDialog } from '@angular/material/dialog';
-import { filter, switchMap, takeUntil, tap } from 'rxjs/operators';
+import { catchError, filter, map, switchMap, takeUntil } from 'rxjs/operators';
 import { GIO_DIALOG_WIDTH, GioConfirmDialogComponent, GioConfirmDialogData } from '@gravitee/ui-particles-angular';
-import { ActivatedRoute, Router } from '@angular/router';
+import { ActivatedRoute, Params, Router } from '@angular/router';
+import { toSignal } from '@angular/core/rxjs-interop';
 
 import {
   ApiDocumentationV4EditFolderDialog,
   ApiDocumentationV4EditFolderDialogData,
 } from './dialog/documentation-edit-folder-dialog/api-documentation-v4-edit-folder-dialog.component';
 
-import { ApiDocumentationV2Service } from '../../../services-ngx/api-documentation-v2.service';
+import { ApiDocumentationPageResult, ApiDocumentationV2Service } from '../../../services-ngx/api-documentation-v2.service';
 import { SnackBarService } from '../../../services-ngx/snack-bar.service';
 import { ApiV2Service } from '../../../services-ngx/api-v2.service';
 import { Api, PageType, EditDocumentationFolder, Breadcrumb, Page, CreateDocumentationFolder } from '../../../entities/management-api-v2';
 
+interface ApiDocumentationV4ListData {
+  pages: Page[];
+  breadcrumbs: Breadcrumb[];
+}
+
 @Component({
   selector: 'api-documentation-v4',
   templateUrl: './api-documentation-v4.component.html',
   styleUrls: ['./api-documentation-v4.component.scss'],
 })
 export class ApiDocumentationV4Component implements OnInit, OnDestroy {
-  private unsubscribe$: Subject<void> = new Subject<void>();
   api: Api;
-  parentId: string;
-  pages: Page[];
-  breadcrumbs: Breadcrumb[];
-  isLoading = false;
```

Comment:
Why did you decide to get rid of `isLoading`?
And how do you manage situation when component is still loading?

Reply:
Using the `| async` means we don't have to use `isLoading`